mailpoet/lib/Statistics/GATracking.php

```diff
@@ -59,10 +70,27 @@ private function addParams($extractedLinks, $gaCampaign, $internalHost = null) {
         // Process only internal links (i.e. pointing to current site)
         continue;
       }
-      list($path, $search, $hash) = $this->splitLink($extractedLink['link']);
-      $search = empty($search) ? $params : $search . '&' . $params;
-      $processedLink = $path . '?' . $search . ($hash ? '#' . $hash : '');
+
       $link = $extractedLink['link'];
+
+      // Do not overwrite existing query parameters
+      $parsedUrl = parse_url($link);
+      $linkParams = $params;
+      if (isset($parsedUrl['query'])) {
+        foreach (array_keys($params) as $param) {
+          if (strpos($parsedUrl['query'], $param) !== false) {
```

Comment:
This will yield a false positive on a query like `foo=utm_source`. Can we check that parameter names are present exactly where parameter names should be? E.g. `strpos($parsedUrl['query'], $param . '=') !== false`.

Reply:
Thank you @alex-mpoet . Good point. https://github.com/mailpoet/mailpoet/pull/5413/commits/d74991607d4535ef1af7e7406cfc1f3778d94bf0 addresses this issue..